addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -559,6 +563,22 @@ export class Wysiwyg extends Component {
         );
 
         this.$editable.on('click', '.o_image, .media_iframe_video', e => e.preventDefault());
+        this.$editable.on('click', '.o_editor_banner_icon', event => {
+            if (this.bannerEmojiPicker.isOpen) {
+                return;
+            }
+            this.bannerEmojiPicker.open(event.target, {
+                onSelect: emoji => {
+                    const icon = document.createElement('i');
+                    icon.innerText = emoji;
+                    icon.classList.add('o_editor_banner_icon', 'mb-3', 'fst-normal', 'cursor-pointer');
+                    if (event.target.getAttribute('aria-label')) {
+                        icon.setAttribute('aria-label', event.target.getAttribute('aria-label'));
+                    }
+                    event.target.replaceWith(icon);
```

Comment:
Hmm, do we have to manually re-create elements, can't we just replace the "innerText" content with the new emoji?

Reply:
I replaced the elements to make sure that we don't have left over classes from the legacy fontawesome icon (like: `fa`, `fa-4` and the class for the actual icon).

Reply:
As your click selector will only work with the new structure and the new class, in which case can we have left over classes? Not sure I'm seeing it.

Last thing to check though, rest is looking good, thanks for the updates 🙂

Reply:
You're absolutely right. The event listener will only be attached to the new banners as the legacy banners will not have any element matching with `.o_editor_banner_icon` (it's the class I added). Therefore, I don't have to transform the legacy icons whatsoever. I updated and simplified the code accordingly :+1: